wily/commands/rank.py

```diff
@@ -0,0 +1,128 @@
+"""
+Rank command.
+
+The report command gives a table of files sorted according their ranking scheme
+of a specified metric.
+Will compare the values between files and return a sorted table.
+
+
+TODO: Refactor individual file work into a separate function
+TODO: Layer on Click invocation in operators section, __main__.py file
+TODO: Is the a better way to retrieve the revision number than via index?
+"""
+import tabulate
+import os
+import operator as op
+
+from wily import logger, format_date, format_revision, MAX_MESSAGE_WIDTH
+from wily.config import DEFAULT_GRID_STYLE
+from wily.operators import resolve_metric, MetricType
+from wily.state import State
+
+
+def rank(config, path, metric="maintainability.index", revision_number=0):
+    """
+
+    :param config: The configuration
+    :type config: :class:'wily.config.WilyConfig'
+
+    :param path: The path to the file
+    :type path ''str''
+
+    :param metric: Name of the metric to report on
+    :type metric: ''str''
+
+    :param revision_number: Version of git repository to revert to.
+    :type revision_number: ''int''
+
+    :return: Sorted table of all files in path, sorted in order of metric.
+    """
+    logger.debug("Running rank command")
+    logger.info(f"-----------Rank for {metric}------------")
+
+    data = []
+
+    operator, key = metric.split(".")
+    metric = resolve_metric(metric)
+    metric_meta = {
```

Comment:
You don't need this :-) use `title = metric.name` later in the file

Reply:
this has been updated in the "headers" section of the new commit - let me know if there was another area where this needed to applied to.

line 109 of second commit:

   headers = ("File", "Revision", "Author", "Date", metric.name)